src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImpl.java

```diff
@@ -1,46 +1,93 @@
 import uk.gov.companieshouse.api.accounts.model.rest.CompanyAccount;
+import uk.gov.companieshouse.api.accounts.repository.CompanyAccountRepository;
 import uk.gov.companieshouse.api.accounts.service.CompanyAccountService;
-import uk.gov.companieshouse.api.accounts.transformer.GenericTransformer;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.accounts.transaction.TransactionManager;
+import uk.gov.companieshouse.api.accounts.transformer.CompanyAccountTransformer;
 
 @Service
-public class CompanyAccountServiceImpl extends
-        AbstractServiceImpl<CompanyAccount, CompanyAccountEntity> implements CompanyAccountService {
+public class CompanyAccountServiceImpl implements CompanyAccountService {
 
     @Autowired
-    public CompanyAccountServiceImpl(
-            @Qualifier("companyAccountRepository") MongoRepository<CompanyAccountEntity, String> mongoRepository,
-            @Qualifier("companyAccountTransformer") GenericTransformer<CompanyAccount, CompanyAccountEntity> transformer) {
-        super(mongoRepository, transformer);
+    private TransactionManager transactionManager;
+
+    @Autowired
+    private CompanyAccountRepository companyAccountRepository;
+
+    @Autowired
+    private CompanyAccountTransformer companyAccountTransformer;
+
+    /**
+     * {@inheritDoc}
+     */
+    public CompanyAccount createCompanyAccount(CompanyAccount companyAccount) {
+        HttpServletRequest request = getRequestFromContext();
+        Transaction transaction = getTransactionFromSession(request);
+
+        String id = generateID();
+        String companyAccountLink = createSelfLink(transaction, id);
+        addKind(companyAccount);
+        addEtag(companyAccount);
+        addLinks(companyAccount, companyAccountLink);
+
+        CompanyAccountEntity companyAccountEntity = companyAccountTransformer.transform(companyAccount);
+
+        companyAccountEntity.setId(id);
+        companyAccountRepository.insert(companyAccountEntity);
+
+        transactionManager.updateTransaction(transaction.getId(),
+                request.getHeader("X-Request-Id"),
+                companyAccountLink);
+
+        return companyAccount;
+    }
+
+    private void addLinks(CompanyAccount companyAccount, String companyAccountLink) {
+        Map<String, String> map = new HashMap<>();
+        map.put(LinkType.SELF.getLink(), companyAccountLink);
+        companyAccount.setLinks(map);
     }
 
-    @Override
-    public void addLinks(CompanyAccount rest) {
+    private String createSelfLink(Transaction transaction, String id) {
+        return getTransactionSelfLink(transaction) + "/company-accounts/" + id;
+    }
+
+    private String getTransactionSelfLink(Transaction transaction) {
+        return transaction.getLinks().get(LinkType.SELF.getLink());
+    }
+    
+    private void addKind(CompanyAccount rest) {
+        rest.setKind("company-accounts");
```

Comment:
this kind should be `company-accounts#company-accounts` as this is what we've decided for the `small-full-accounts#small-full-accounts`